STAT/AliDrawStyle.cxx

```diff
@@ -752,39 +752,53 @@ Float_t AliDrawStyle::PrepareValue(const char* styleName, TString propertyName,
-TObjArray *AliDrawStyle::ReadCSSFile(const char *  inputName, TObjArray * cssArray, Int_t verbose) {
-  //check file exist
-  if (gSystem->GetFromPipe(TString("[ -f ") + TString(inputName) +  TString(" ] && echo 1 || echo 0")) == "0") {
-    ::Error("AliDrawStyle::ReadCSSFile","File %s doesn't exist", inputName);
-    return nullptr;
-  }
-  TString inputCSS = gSystem->GetFromPipe(TString::Format("cat %s",inputName).Data());     // I expect this variable is defined
-  //remove comments:
+//TODO: add test for this @Boris
+///
+/// \param inputCSS
+/// \param cssArray
+/// \param verbose
+/// \return
+TObjArray *AliDrawStyle::ReadCssString(TString inputCSS, TObjArray *cssArray, Int_t verbose) {
   while (inputCSS.Index("*/") > 0) {
-    inputCSS = inputCSS(0, inputCSS.Index("/*")) + inputCSS(inputCSS.Index("*/")+2, inputCSS.Length());
+    inputCSS = TString(inputCSS(0, inputCSS.Index("/*"))) + TString(inputCSS(inputCSS.Index("*/")+2, inputCSS.Length()));
   }
-  //inputCSS.ReplaceAll("\n", ""); //  check performance and implement better variant;
+  inputCSS.ReplaceAll("\n", ""); //  check performance and implement better variant;
   TObjArray *tokenArray = inputCSS.Tokenize("{}");   //assuming we can not use {} symbols in the style IDS
   Int_t entries = tokenArray->GetEntries();
   if (cssArray==nullptr) {
     cssArray = new TObjArray(entries / 2);
   }
+  if (verbose == 4)
+    ::Info("AliDrawStyle::ReadCSSString","In input CSS string %s was parsed to:\n", inputCSS.Data());
   for (Int_t i = 0; i < entries; i += 2) {
     if (i + 1 >= entries) continue;
     TString selector = tokenArray->At(i)->GetName();
     TString declaration = tokenArray->At(i + 1)->GetName();
+    if (verbose == 4)
+      printf("selector: %s\ndeclaration: %s\n", selector.Data(), declaration.Data());
     cssArray->AddLast(new TNamed(selector.Data(), declaration.Data()));
   }
   return cssArray;
 }
 
+/// Read CSS html like files  (*see also AliRoot modification in CSS)
+/// TODO:
+/// * proper exception  handling (Boris)
+///   * Use ::Error verbosity according debug level
+/// \param inputName     - input file to read
+/// \param verbose       - specify verbose level for ::error and ::info (Int_t should be interpreted as an bit-mask)
+/// \return              - TObjArray  with the pairs TNamed of the CSS <Selector, declaration> or  TObjArray (recursive structure like includes)
+TObjArray *AliDrawStyle::ReadCSSFile(const char *  inputName, TObjArray * cssArray, Int_t verbose) {
+  //check file exist
+  if (gSystem->GetFromPipe(TString("[ -f ") + TString(inputName) +  TString(" ] && echo 1 || echo 0")) == "0") {
+    ::Error("AliDrawStyle::ReadCSSFile","File %s doesn't exist", inputName);
+    return nullptr;
+  }
+
+  TString inputCSS = gSystem->GetFromPipe(TString::Format("cat %s",inputName).Data());     // I expect this variable is defined
```

Comment:
What the HELL is this? Use standard C to open a file, for God's sake.

Reply:
I will let Boris to change in case.  Reading full  ascii file as an TString using c++ needs several  lines. 
````
std::ifstream t(inputName);
std::stringstream buffer;
buffer << t.rdbuf();
....
````

Reply:
Marian, actually it's your code xD